### DISCUSSION TITLE
Freeplane community online conference on Saturday, July 23, 2022

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#473] [Announcements]
Join us for the third roundtable to talk about anything related to Freeplane. Your curiosity sets up the agenda. You can ask any questions or present any idea or experience in a short lightning talk. The Lean Coffee meeting format allows us to be flexible and to talk about everything that most interests you as long as needed. In the past, our meetings helped us to get to know each other, exchange ideas, recharge our batteries, and join our efforts in making and documenting Freeplane.

A very quick video introduction to Lean Coffee is available at https://www.youtube.com/watch?v=4TOOdAXwcrk .

We meet at an exclusive online location https://spatial.chat/s/freeplane.
The online service allows only 25 people to join, and I hope it is the right number.

Time: Saturday, July 23, 2022
13:00 London
14:00 Amsterdam, Berlin, Rome, Stockholm, Vienna
online, https://spatial.chat/s/freeplane
The planned event duration is 2 hours.

[Download calendar file](https://github.com/freeplane/freeplane/files/8998625/freeplane-conf-July.ics.zip)

There is no registration for the event.
However, let us know by writing a comment if you going to join us.

See you!
Dimitry

**Comment by dpolivaev** [comment]
Hello folks, next Saturday we are going to meet online again.
It would be nice if you could share here if you are going to join us.

Cheers,
Dimitry

**Comment by 1-DimensionalBox** [comment]
Yes, I could join! 👍

**Comment by sifran-github** [comment]
i will join

**Reply by 1-DimensionalBox** [reply]
Great, then we should talk scripting documentation! 🙂

**Comment by nnako** [comment]
I will be present. looking forward to seeing you there (don't forget to prepare your cameras, at least for the introduction round, please).

**Comment by 1-DimensionalBox** [comment]
Thank you for a pleasant community conference! You can now find the meeting minutes as a mindmap:

[EDITED based upon comments by dpolivaev and sifran-github]

* [Here, packed in a .zip archive](https://github.com/freeplane/freeplane/files/9193620/Freeplane.online.community.conference.23.July.2022.-.minutes.v2.zip)
* [Here, exported to .pdf](https://github.com/freeplane/freeplane/files/9193622/Freeplane.online.community.conference.23.July.2022.-.minutes.v2.pdf)

**Reply by 1-DimensionalBox** [reply]
Also, even though I could not get some things right with formatting, here below you can find an export to HTML:

[EDITED based upon comments by dpolivaev and sifran-github]

<br><!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
  <head>
  </head>
  <body>
    <p>Freeplane (online) community conference, 23 July 2022 - minutes</p>
    <ul>
      <li><span style="font-weight: bold; ">nnako: hackathon</span>
        <ul>
          <li>nnako: developer hangout meeting
            <ul>
              <li>social event</li>
              <li>coding together</li>
            </ul>
          </li>
          <li>dpolivaev (= DP): usefulness depends on working style
            <ul>
              <li>"I prefer to code by myself"</li>
              <li>"but I imagine others might appreciate it"</li>
            </ul>
          </li>
          <li>will invite those interested via freeplane/freeplane discussion
            forum</li>
        </ul>
        <br>
      </li>
      <li><span style="font-weight: bold; ">the progress in documentation was
          appreciated by all participants!</span>
        <ul>
          <li>🙏 @dpolivaev</li>
          <li>🙏 @macmarrum</li>
          <li>🙏 @ofossum</li>
        </ul>
        <br>
      </li>
      <li><span style="font-weight: bold; ">1-DimensionalBox (= 1DB): scripting
documentation</span></li>
<br>
      <ul>
        <li><span style="font-weight: bold; ">DP: structure of attic</span>
          <ul>
            <li><a href="[https://docs.freeplane.org/#/attic/new-wiki-initiative/customizing-freeplane]%28view-source:https://docs.freeplane.org/#/attic/new-wiki-initiative/customizing-freeplane%29"
                target="_blank">New wiki initiative</a>
              <ul>
                <li>new pages in dev</li>
              </ul>
            </li>
            <li><a href="[https://docs.freeplane.org/#/attic/old-mediawiki-content/Home]%28view-source:https://docs.freeplane.org/#/attic/old-mediawiki-content/Home%29"
                target="_blank">Old MediaWiki content</a>
              <ul>
                <li>Old MediaWiki site converted to docsify</li>
              </ul>
            </li>
            <li><a href="[https://www.freeplane.org/oldwiki]%28view-source:https://www.freeplane.org/oldwiki%29"
                target="_blank">Old MediaWiki site</a>
              <ul>
                <li>archived MediaWiki site</li>
              </ul>
            </li>
          </ul>
          <br>
        </li>
        <li><span style="font-weight: bold; ">sifran: info about scripting is
            incomplete</span>
          <ul>
            <li>"I would like to add info about API"</li>
          </ul>
          <br>
        </li>
        <li><span style="font-weight: bold; ">add info about Groovy programming?</span>
          <ul>
            <li>sifran &amp; 1DB: no, keep focus on Freeplane</li>
          </ul>
          <br>
        </li>
        <li><span style="font-weight: bold; "><a href="[https://docs.freeplane.org/#/scripting/Scripts_collection]%28view-source:https://docs.freeplane.org/#/scripting/Scripts_collection%29"
              target="_blank">Scripts collection: have people sign scripts?</a>
          </span>
          <ul>
            <li>1DB: as it is now, things are potentially insecure</li>
            <li>DP
              <ul>
                <li>
                  <p> this would be unpractical </p>
                </li>
                <li>instead, add an extra warning</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
      <br>
      <li><span style="font-weight: bold; ">1DB: docs: how to edit</span></li>
      <br>
<ul>
          <li><span style="font-weight: bold; ">DP: ensemble programming</span>
            <ul>
              <li>
                <ul>
                  <li>one central programmer, others follow through video
                    sharing</li>
                  <li>alternatively, closely parallel programming, updating
                    every 5 min.s</li>
                </ul>
              </li>
              <li>"the advantage is that working together allows for big
                decisions to be taken. Otherwise, we tend to put these decisions
                of."</li>
              <li>will invite those interested via freeplane/docs discussion
                forum</li>
            </ul>
            <br>
          </li>
          <li><span style="font-weight: bold; ">issues vs discussions</span>
            <ul>
              <li>
                <ul>
                  <li>issues should regards specific changes in
                    code/documentation
                    <ul>
                      <li>sifran: "but how about a bug? That can be quite
                        general, but I would put up an issue for it."</li>
                    </ul>
                  </li>
                  <li>
                    <p> discussions can be more general </p>
                  </li>
                </ul>
              </li>
            </ul>
            <br>
          </li>
        </ul>
      </li>
      <li><span style="font-weight: bold; ">EdoFro: Tutorial-o-Matic</span>
        <ul>
          <li>a language for writing tutorials, building upon Freeplane</li>
          <li>EdoFro: "please look into its tutorial" ;-)</li>
        </ul>
      </li>
      <li><span style="font-weight: bold; ">mobile phone</span>
        <ul>
          <li>1DB:
            <ul>
              <li>use FOSS app like orgzly (org-mode)
                <ul>
                  <li>matches mindmap organization, but organized as text file</li>
                  <li>fnatter: have worked with importing org-mode</li>
                </ul>
              </li>
              <li>combine with syncing app, like Syncthing</li>
            </ul>
          </li>
          <li>sifran &amp; DP use SimpleMind
            <ul>
              <li>allows for FreeMind importing</li>
            </ul>
          </li>
          <li>to be continued in the discussion forum</li>
        </ul>
        <br>
      </li>
      <li><span style="font-weight: bold; ">DP: any requests for features?</span></li>
        <ul>
          <li><span style="font-weight: bold; ">fnatter: more stable version,
              like an LTS?</span>
            <ul>
              <li>DP: I cannot commit to this
                <ul>
                  <li>prereleases: tested by ~10 developers</li>
                  <li>stable: used by ~1000 users</li>
                </ul>
              </li>
            </ul>
            <br>
          </li>
        </ul>
      </li>
        <ul>
          <li><span style="font-weight: bold; ">1DB: toolbar icons: customize</span>
            <ul>
              <li>DP: already possible</li>
              <li>
                <p> find information <a href="[https://www.freeplane.org/oldwiki/index.php/Menus_and_commands#Example_of_changes_than_can_be_performed_in_the_Freeplane_menu_file]%28view-source:https://www.freeplane.org/oldwiki/index.php/Menus_and_commands#Example_of_changes_than_can_be_performed_in_the_Freeplane_menu_file%29">here</a></p>
              </li>
            </ul>
            <br>
          </li>
          <li><span style="font-weight: bold; ">sifran: running Freeplane from within the PortableApps does not work </span>
            <ul>
              <li>DP: I think too few users would be helped by this, so I cannot
                prioritize it</li>
              <li>sifran & DP agree: currently no elegant solution</li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </body>
</html>

**Reply by dpolivaev** [reply]
> DP: issues should be assigned to specific developers, otherwise they are not implemented

It is not what I was trying to say. I even do not know where this idea comes from.
I could have said that for each concrete trackable piece of work we need an issue because discussions are too general and not trackable.

**Reply by ofossum** [reply]
Sadly i couldn't participate, so thank a lot @1-DimensionalBox for the helpful protocol!

For coding together one could use some terminal multiplexer as [screen](https://www.gnu.org/software/screen/) and [tmux](https://github.com/tmux/tmux/wiki/Getting-Started) for shared terminal-sessions - in case you use linux.
Otherwise and more user-friendly, one could just use the "Code" module from [CryptPad](https://cryptpad.fr/) (there is an [open instance hosted from german NGO "Digitalcourage"](https://cryptpad.digitalcourage.de/) among others).
You can code there like in an Etherpad, have a full timeline what was written and also use syntax-highlighting for most programming-languages. ;)

**Reply by 1-DimensionalBox** [reply]
Thanks for looking over the minutes, @dpolivaev. I seem to have misunderstood you, but I will correct this.

**Reply by 1-DimensionalBox** [reply]
I have corrected the mindmap and removed the quote :heavy_check_mark:

**Comment by sifran-github** [comment]
nice meeting. thank you for the summary.

just a quick note - I did not offer to include java with the portableapps version, i just described the issue that: running freeplane (the portableapps version) not from within the protableapps launcher does not work and requires manual intervention. described here: https://github.com/freeplane/freeplane/discussions/462#discussioncomment-2988852

Dimitry said that some people suggested to include java but he thinks that this is not a good idea, to which I agreed. Dimitry and I agreed that currently we do not have an elegant solution to it.

Now that I think about it again, if freeplane in the zip version is "portable" anyway and doesn't write to system files, I might just use the zip version instead of the portableapps one.

**Reply by 1-DimensionalBox** [reply]
Thank you for your reply to the minutes! I will add extra info to the minutes (and link to this comment of yours).

**Reply by 1-DimensionalBox** [reply]
I have now added the extra info to the minutes ✅

**Comment by nnako** [comment]
Hey folks,

somehow, we forgot that we wanted to schedule the next Freeplane online user conference for TODAY, 1400 German time. There seems to be no participant, though, following the link of the old meeting...

Have I been mistaken?
